test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -248,6 +248,140 @@ def test_addding_individual_bit(self):
         self.assertEqual(list(dag.qregs.values()), [qr])
 
 
+class TestDagWireRemoval(QiskitTestCase):
+    """Test removal of registers and idle wires."""
+
+    def setUp(self):
+        super().setUp()
+        self.dag = DAGCircuit()
+        qreg = QuantumRegister(3, "qr")
+        creg0 = ClassicalRegister(2, "c0")
+        creg1 = ClassicalRegister(2, "c1")
+        creg2 = ClassicalRegister(name="c2", bits=list(creg1))
+        clbit = Clbit()
+        self.dag.add_qreg(qreg)
+        self.dag.add_creg(creg0)
+        self.dag.add_creg(creg1)
+        self.dag.add_creg(creg2)
+        self.dag.add_clbits([clbit])
+
+        self.qreg = qreg
+        self.creg0 = creg0
+        self.creg1 = creg1
+        self.creg1_alias = creg2
+        self.individual_clbit = clbit
+
+        self.original_cregs = [self.creg0, self.creg1, self.creg1_alias]
+
+        # skip adding bits of creg1_alias since it's just an alias for creg1
+        self.original_clbits = [
+            b for reg in self.original_cregs if reg != self.creg1_alias for b in reg
+        ] + [self.individual_clbit]
+
+    def assert_cregs_is(self, cregs, excluding=None):
+        """Assert test DAG cregs match the expected values.
+
+        Args:
+            cregs (Iterable(ClassicalRegister)): the classical registers to expect
+            excluding (Set(ClassicalRegister)): classical registers to remove from
+            ``cregs`` before the comparison."""
+        if excluding is None:
+            excluding = {}
```

Comment:
Probably not important, but the documentation says `excluding` should be a set, and this creates a dict.

(Comment repeats for the other test.)

Reply:
Done in f717f58.